### ISSUE TITLE
Incorrect documentation warnings for references to positional record types and parameters

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn#46550]
**Version Used**: VS 16.7-p6

```cs
namespace SomeNamespace
{
    /// <summary>
    /// ⚠ CS1574 XML comment has cref attribute 'R' that could not be resolved
    ///                              ↓
    /// Documentation for <see cref="R"/>.
    /// </summary>
    /// ⚠ CS1572 XML comment has a param tag for 'A', but there is no parameter by that name
    /// ⚠ CS1734 XML comment on 'R' has a paramref tag for 'A', but there is no parameter by that name
    ///              ↓                                            ↓
    /// <param name="A">Description for parameter <paramref name="A"/></param>
    public record R(int A);
}
```

**Comment by jaredpar** [comment]
Using #44571 as the primary discussion issue for this problem.
